django/http/request.py

```diff
@@ -249,28 +260,28 @@ def _load_post_and_files(self):
             self._mark_post_parse_error()
             return
 
-        if self.META.get('CONTENT_TYPE', '').startswith('multipart/form-data'):
-            if hasattr(self, '_body'):
-                # Use already read data
-                data = BytesIO(self._body)
-            else:
-                data = self
-            try:
+        try:
+            if self.META.get('CONTENT_TYPE', '').startswith('multipart/form-data'):
+                if hasattr(self, '_body'):
+                    # Use already read data
+                    data = BytesIO(self._body)
+                else:
+                    data = self
                 self._post, self._files = self.parse_file_upload(self.META, data)
-            except MultiPartParserError:
-                # An error occurred while parsing POST data. Since when
-                # formatting the error the request handler might access
-                # self.POST, set self._post and self._file to prevent
-                # attempts to parse POST data again.
-                # Mark that an error occurred. This allows self.__repr__ to
-                # be explicit about it instead of simply representing an
-                # empty POST
-                self._mark_post_parse_error()
-                raise
-        elif self.META.get('CONTENT_TYPE', '').startswith('application/x-www-form-urlencoded'):
-            self._post, self._files = QueryDict(self.body, encoding=self._encoding), MultiValueDict()
-        else:
-            self._post, self._files = QueryDict('', encoding=self._encoding), MultiValueDict()
+            elif self.META.get('CONTENT_TYPE', '').startswith('application/x-www-form-urlencoded'):
+                self._post, self._files = QueryDict(self.body, encoding=self._encoding), MultiValueDict()
+            else:
+                self._post, self._files = QueryDict(encoding=self._encoding), MultiValueDict()
+        except (MultiPartParserError, SuspiciousOperation):
+            # An error occured while parsing POST data. Since when
+            # formatting the error the request handler might access
+            # self.POST, set self._post and self._file to prevent
+            # attempts to parse POST data again.
+            # Mark that an error occured. This allows self.__repr__ to
+            # be explicit about it instead of simply representing an
+            # empty POST
+            self._mark_post_parse_error()
+            raise
```

Comment:
I don't think using `six.reraise` is necessary here. The code just calls `self._mark_post_parse_error()` and raises same two exceptions. I'd use `six.reraise` if the code looks like

```py
except (MultiPartParserError, SuspiciousOperation) as exc:
    # ...
    raise OtherException(six.text_type(exc))
```

or

```py
except MultiPartParserError:
    # ...
    raise MultiPartParserError('custom message')
except  SuspiciousOperation:
    # ...
    raise SuspiciousOperation('custom message')
```